### PR TITLE
29. Authentication

### PR DESCRIPTION
**Pull request by rajuTK9**

### CODE REVIEW

**Comment by rajuTK9** [comment]
I have successfully set up authentication for users and administrators.

User's progress data is also now saving to my data base.

File: client/src/App.js [hunk 1]
```diff
@@ -9,22 +9,34 @@ import Notes from "./components/Notes/Notes";
...
 import NotesIcon from "./components/NotesIcon/NotesIcon";
 import CourseUpdate from "./components/Admin/CourseUpload/CourseUpdate";
+import { Auth0Provider } from "@auth0/auth0-react";
+import Protected from "./Protected";
 
 function App() {
   return (
-    <>
+    <Auth0Provider
+      domain="dev-ynlvngvzhos0c4m0.us.auth0.com"
```
Comment (aditya-190) [inline client/src/App.js]: These values can be added in the .env files.

File: client/src/Protected.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+import { Outlet } from "react-router";
+import GetUser from "./data/GetUser";
+
+export default function Protected() {
+  const user = GetUser();
+  if (user) console.log(user.isAdmin);
```
Comment (aditya-190) [inline client/src/Protected.js]: It is recommended to remove all the console.log from the code before merging them into your main code.

**Review by aditya-190 — CHANGES_REQUESTED** [review]
Please remove all the console.log's & add all credentials to .env files before merging.

File: client/src/Protected.js (hunk 2 above)
Comment (rajuTK9) [reply client/src/Protected.js]: removed!

File: client/src/App.js (hunk 1 above)
Comment (rajuTK9) [reply client/src/App.js]: done!

**Pull request merged**